cloud/softlayer/sl.py

```diff
@@ -0,0 +1,430 @@
+version_added: "2.1"
+options:
+  instance_id:
+    description:
+      - Instance Id of the virtual instance to perform action option
+    required: false
+    default: null
+  hostname:
+    description:
+      - Hostname to be provided to a virtual instance
+    required: false
+    default: null
+  domain:
+    description:
+      - Domain name to be provided to a virtual instance
+    required: false
+    default: null
+  datacenter:
+    description:
+      - Datacenter for the virtual instance to be deployed
+    required: false
+    default: null
+  tags:
+    description:
+      - Tag or list of tags to be provided to a virtual instance
+    required: false
+    default: null
+  hourly:
+    description:
+      - Flag to determine if the instance should be hourly billed
+    required: false
+    default: true
+  private:
+    description:
+      - Flag to determine if the instance should be private only
+    required: false
+    default: false
+  dedicated:
+    description:
+      - Falg to determine if the instance should be deployed in dedicated space
+    required: false
+    default: false
+  local_disk:
+    description:
+      - Flag to determine if local disk should be used for the new instance
+    required: false
+    default: true
+  cpus:
+    description:
+      - Count of cpus to be assigned to new virtual instance
+    required: true
+    default: null
+  memory:
+    description:
+      - Amount of memory to be assigned to new virtual instance
+    required: true
+    default: null
+  disks:
+    description:
+      - List of disk sizes to be assigned to new virtual instance
+    required: true
+    default: [25]
+  os_code:
+    description:
+      - OS Code to be used for new virtual instance
+    required: false
+    default: null
+  image_id:
+    description:
+      - Image Template to be used for new virtual instance
+    required: false
+    default: null
+  nic_speed:
+    description:
+      - NIC Speed to be assigned to new virtual instance
+    required: false
+    default: 10
+  public_vlan:
+    description:
+      - VLAN by its Id to be assigned to the public NIC
+    required: false
+    default: null
+  private_vlan:
+    description:
+      - VLAN by its Id to be assigned to the private NIC
+    required: false
+    default: null
+  ssh_keys:
+    description:
+      - List of ssh keys by their Id to be assigned to a virtual instance
+    required: false
+    default: null
+  post_uri:
+    description:
+      - URL of a post provisioning script ot be loaded and exectued on virtual instance
+    required: false
+    default: null
+  state:
+    description:
+      - Create, or cancel a virtual instance. Specify "present" for create, "absent" to cancel.
+    required: false
+    default: 'present'
+  quantity:
+    description:
+      - Quantity of virtual instances to provision
+    required: false
+    default: 0
+  increment:
```

Comment:
Making edits to remove this functionality within the module. Makes better sense to utilize it outside.